stablehlo/reference/Ops.cpp

```diff
@@ -1589,6 +1594,11 @@ Tensor dynamicIotaOp(Axis iotaDimension, const Tensor &outputShape,
       "dynamic result types are not supported at the moment");
 }
 
+Tensor dynamicReshapeOp(const Tensor &operand, const Tensor &outputShape,
+                        ShapedType resultType) {
+  return reshapeOp(operand, resultType);
```

Comment:
Can we add a check at https://github.com/openxla/stablehlo/blob/8ba7728d3fdc3ea882e893ee7e53255c95ee0e5a/stablehlo/reference/Ops.cpp#L310 to error on dynamic output shape. 

Also remove the error at https://github.com/openxla/stablehlo/blob/8ba7728d3fdc3ea882e893ee7e53255c95ee0e5a/stablehlo/reference/Ops.cpp#L1588

Reply:
https://github.com/openxla/stablehlo/issues/2294 will fix it. Before I merge reshape op PR.